### DISCUSSION TITLE
Filter DataTable on array

### DISCUSSION THREAD

**Discussion by ratracegrad** [primefaces/community#1831] [PrimeVue]
My Data looks like this:
[
  { 
    name: 'Jennifer',
    countries: [
     {name: 'England', visited: true },
     { name: 'Scotland', visited: true}
   ]
  }
]

I want to be able to filter where 'countries.name' is equal to 'England'

my filters:
const filters = ref({
  global: { value: null, matchMode: FilterMatchMode.CONTAINS },
  'countries.name': { value: null, matchMode: FilterMatchMode.IN },
})

in my datatable I have
<DataTable
  v-model:filters="filters"
  :global-filter-fields="['countries.name']
>

When I select a country like 'England' then the table shows nothing. 
Any suggestion on what I am doing wrong?

**Comment by mertsincan** [comment]
Hi @ratracegrad,

Could you share a stackblitz link for us? Your 'countries' data is an array and I think you made some custom changes while displaying the nested array in the DataTable. So a sample code would be good for us.

**Reply by ratracegrad** [reply]
Here is a stackblitz example. 
Each entry in the dataTable has a field called `types` which is an array of strings.
In the filter, you can select one or more types.  What I want is for it filter if any of the selected types are in the `types` array.

https://stackblitz.com/edit/5cwejm?file=src%2FApp.vue

**Reply by ratracegrad** [reply]
If I set the filter to:
```types: { value: null, matchMode: FilterMatchMode.IN },```
it will not filter for any selection.

If I set the filter to:
```types: { value: null, matchMode: FilterMatchMode.CONTAINS },```
It will match for a single selection but fails if I select a second selection. I want it to be able to filter on one or more selections that match any value in the types array.

**Reply by iPOiZN** [reply]
Did you solve this problem somehow?

**Comment by iPOiZN** [comment]
+1
